monix-execution/shared/src/main/scala/monix/execution/FutureUtils.scala

```diff
@@ -115,18 +116,28 @@ object FutureUtils {
     *
     * Similar to `Future.transformWith` from Scala 2.12.
     */
-  def transformWith[A,B](source: Future[A], f: Try[A] => Future[B])(implicit ec: ExecutionContext): Future[B] = {
+  def transformWith[A,B](source: Future[A], f: Try[A] => Future[B])(implicit ec: ExecutionContext): CancelableFuture[B] = {
```

Comment:
Chiming in since asked on twitter https://twitter.com/ktosopl/status/892221038242570240 

This is binary incompatible, MiMa is right.
Changing the type like this will cause `MethodNotFound`.

Simple way to test those things if you think MiMa may be mistaken:

```scala
$ ls *scala
it.scala    lol.scala   main.scala
ktoso @ 三日月/tmp
$ cat *scala
object f { def f(): A = null }

trait A
trait B extends A

object Main extends App {

 val a =   f.f()
}
ktoso @ 三日月/tmp
$ scalac *.scala
ktoso @ 三日月/tmp
$ java -cp /Users/ktoso/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.12.2.jar:. Main
ktoso @ 三日月/tmp
$ vim it.scala
ktoso @ 三日月/tmp
$ scalac it.scala
ktoso @ 三日月/tmp
$ cat it.scala
object f { def f(): B = null }

ktoso @ 三日月/tmp
$ java -cp /Users/ktoso/.ivy2/cache/org.scala-lang/scala-library/jars/scala-library-2.12.2.jar:. Main
Exception in thread "main" java.lang.NoSuchMethodError: f$.f()LA;
```

Hope this helps